tests/install/test_distribution.pm

```diff
@@ -9,7 +9,24 @@ sub run {
     diag('initialize working copy of openSUSE tests distribution with correct user');
     assert_script_run('username=bernhard email=bernhard@susetest /usr/share/openqa/script/fetchneedles', 3600);
     save_screenshot;
-    assert_script_run('zypper --no-refresh -n in os-autoinst-distri-opensuse-deps', 600);
+
+    # 104 is the zypper exit code if a package cannot be found
+    my $cmd = <<'EOM';
+(
+    for i in {1..7}
+    do
+        zypper --non-interactive install os-autoinst-distri-opensuse-deps && exit 0
+        rc=$?
+        (( $rc == 104 )) || break
+    done
+    exit $rc
+)
+EOM
+    chomp $cmd;
+    my $ret = assert_script_run(
+        $cmd,
+        timeout => 600,
+    );
```

Comment:
A more common style – at least in our test code – is to write `assert_script_run $cmd, 600;`

And the variable assignment to $ret` is useless for two reasons:
1. You don't use `$ret`
2. `assert_script_run` will throw an exception if the assertion does not hold so use without return code evaluation.

Reply:
The documentation for `assert_script_run` is:
```
assert_script_run($cmd [, timeout => $timeout] [, fail_message => $fail_message] [,quiet => $quiet]);
```
AFAIK passing timeout as a second arg is using the old API which is still supported.

Reply:
Using `$ret` was a leftover from experimenting, will remove it.

Reply:
There are no plans to remove the positional argument but you are right, it says "Deprecated mode".